src/owncloud/autoconfig.php

```diff
@@ -3,23 +3,26 @@
 $snap_name = getenv('SNAP_NAME');
 $snap_developer = getenv('SNAP_DEVELOPER');
 if ($snap_developer == '') {
-        $snap_developer = getenv('SNAP_ORIGIN');
+	$snap_developer = getenv('SNAP_ORIGIN');
 }
 
 $snap_fullname = $snap_name . '.' . $snap_developer;
 
-$data_path = '/var/lib/snaps/'.$snap_fullname.'/current';
+$data_path = '/var/lib/snaps/' . $snap_fullname . '/current';
 
-$AUTOCONFIG = array(
-'directory' => $data_path.'/owncloud/data',
+$writable = getenv('SNAP_DATA');
+$owncloud_password = trim(file_get_contents($writable . '/mysql/owncloud_password'));
```

Comment:
Fetching the DB password from the file directly instead of using environment variables